config-connector/solutions/networking/helm/autoneg/README.md

```diff
@@ -0,0 +1,142 @@
+
+1. Configure clusters and load balancing resources with Helm
+
+    ```bash
+    # validate your chart
+    helm lint ./lb/ --set projectId=[PROJECT_ID]
+
+    # check the output of your chart
+    helm template ./lb/ --set projectId=[PROJECT_ID]
+
+    # install your chart
+    helm install ./lb/ --set projectId=[PROJECT_ID] --generate-name
+    ```
+
+1. Wait for clusters to be created
+
+    ```bash
+    # The command uses cluster names based on the values passed in the previous step
+    kubectl wait --for=condition=Ready containercluster/cluster-na containercluster/cluster-eu
+    ```
+
+1. Configure first cluster
+
+    ```bash
+    # switch the context to the first cluster. The command uses cluster name and zone based on the values used to create the clusters.
+    gcloud container clusters get-credentials cluster-na --zone=us-central1-a
+
+    # validate your chart
+    helm lint ./cluster/ --set projectId=[PROJECT_ID] --set localMessage="Hello from North America!"
+
+    # install your chart
+    helm install ./cluster/ --set projectId=[PROJECT_ID] --set localMessage="Hello from North America!" --generate-name
+
+     # annotate service account
+    kubectl annotate sa -n autoneg-system default iam.gke.io/gcp-service-account=autoneg-system@[PROJECT_ID].iam.gserviceaccount.com
+
+    # ensure pods are ready
+    kubectl wait --for=condition=Ready pods --all
+    ```
+
+1. Repeat the step for the other cluster
+
+    ```bash
+    # switch the context to the second cluster. The command uses cluster name and zone based on the values used to create the clusters.
+    gcloud container clusters get-credentials cluster-eu --zone=europe-west2-a
+
+    # validate your chart
+    helm lint ./cluster/ --set projectId=[PROJECT_ID] --set localMessage="Hello from Europe"
+
+
+    # install your chart
+    helm install ./cluster/ --set projectId=[PROJECT_ID] --set localMessage="Hello from Europe" --generate-name
+
+     # annotate service account
+    kubectl annotate sa -n autoneg-system default iam.gke.io/gcp-service-account=autoneg-system@[PROJECT_ID].iam.gserviceaccount.com
+
+    # ensure pods are ready
+    kubectl wait --for=condition=Ready pods --all
+    ```
+
+1. Switch the context to the main cluster and run verify that multi-cluster ingress is configured
+
+    ```bash
+    # switch the context to the main cluster
+    gcloud container clusters get-credentials [CLUSTER NAME] --zone=[CLUSTER ZONE]
+
+    # curl the external address of the forwarding rule. Note that it might take around 5-10 minutes for load balancing to start working.
+    # You will see the message ("Hello from North America" or "Hello from Europe" based on your location).
+    curl $(kubectl get  computeforwardingrule -o=jsonpath='{.items[0].spec.ipAddress.addressRef.external}')
```

Comment:
I got the following error message:
```
$ curl $(kubectl get  computeforwardingrule -o=jsonpath='{.items[0].spec.ipAddress.addressRef.external}') 

<html><head>
<meta http-equiv="content-type" content="text/html;charset=utf-8">
<title>502 Server Error</title>
</head>
<body text=#000000 bgcolor=#ffffff>
<h1>Error: Server Error</h1>
<h2>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.</h2>
<h2></h2>
</body></html>
```

When I only did the kubectl command, it succeeded:
```
$ kubectl get  computeforwardingrule -o=jsonpath='{.items[0].spec.ipAddress.addressRef.external}'
34.120.245.118
```
Did I do anything wrong?

Reply:
I added two steps to readme to check along the way: the first for each of the services in cluster:
```
check the service and ensure that `anthos.cft.dev/autoneg-status` annotation is present in the output
    get svc node-app-backend -o=jsonpath='{.metadata.annotations}'
```
and the second one on the backend service:
```
gcloud compute backend-services describe node-app-backend-service
```